panc/src/main/scripts/panlint

```diff
@@ -0,0 +1,287 @@
+    if DEBUG:
+        label = 'DEBUG: %04d %-12s |' % (line_number, '...')
+        print Style.DIM + Fore.CYAN + label + Style.RESET_ALL + line
+
+
+def debug_ignored_line(line, line_number):
+    """Print debug information for an ignored line of an input file"""
+    if DEBUG:
+        label = 'DEBUG: %04d %-12s |' % (line_number, 'Ignored')
+        print Fore.CYAN + Style.DIM + label + Fore.RESET + line + Style.RESET_ALL
+
+
+def debug_range(start, end, label, problem=False):
+    """Print debug information referring to a range of characters in a single line"""
+    if DEBUG:
+        label = ('DEBUG: ^^^^ %-12s |' % label)
+        diagnosis = diagnose(start, end)
+        color = Fore.CYAN
+        if problem:
+            color = Fore.RED
+        print Style.DIM + Fore.CYAN + label + Style.BRIGHT + color + diagnosis + Style.RESET_ALL
+
+
+def diagnose(start, end):
+    """Format a line of diagnosis markers from a range of character positions"""
+    return (' ' * start) + ('^' * (end - start))
+
+
+def print_report(filename, line_number, line, diagnosis, message, vi=False):
+    """Print a full report of all problems found with a single line of a processed file"""
+    print
+    print print_fileinfo(filename, line_number, message, vi=vi)
+    print print_line(line)
+    print print_diagnosis(diagnosis)
+
+
+def get_string_ranges(line):
+    """Find all ranges of strings within a single line of text"""
+    string_ranges = []
+    strings = RE_STRING.finditer(line)
+
+    if strings:
+        for string in strings:
+            s, e = string.span()
+            debug_range(s, e, 'String Range')
+            string_ranges.append((s, e))
+
+    return string_ranges
+
+
+def filestats_table(problem_stats):
+    """Return a formatted table of problem counts per file"""
+    t = PrettyTable(['Filename', 'Problems'])
+    t.align['Filename'] = 'l'
+    t.align['Problems'] = 'r'
+    for filename, problem_count in problem_stats.iteritems():
+        if problem_count > 0:
+            t.add_row([filename, problem_count])
+    t.sortby = 'Filename'
+    return t
+
+
+def main():
+    """Main function"""
+    parser = argparse.ArgumentParser(description='Linter for the pan language')
+    parser.add_argument('paths', metavar='PATH', type=str, nargs='+', help='Paths of files to check')
+    parser.add_argument('--ide', action='store_true', help='Output machine-readable results for use by IDEs')
+    parser.add_argument('--vi', action='store_true', help='Output line numbers in a vi option style')
+    parser.add_argument('--debug', action='store_true', help='Enable debug output')
+    parser.add_argument('--table', action='store_true', help='Display a table of per-file problem stats')
+    args = parser.parse_args()
+
+    # Only output colors sequences if the output is a terminal
+    colorama_init(strip = (not stdout.isatty()) or args.ide)
+    global DEBUG
+    DEBUG = args.debug and (not args.ide)
```

Comment:
Based on my experience with argparse, I'd say the same: argument compatibility cannot be described inside the parser, they have to be handled through tests....This may have evolved but this was at least the case in Python 2.4...